image-rs/src/unpack.rs

```diff
@@ -83,6 +88,54 @@ pub fn unpack<R: io::Read>(input: R, destination: &Path) -> Result<()> {
     Ok(())
 }
 
+fn convert_whiteout(path: &Path, header: &Header, destination: &Path) -> Result<bool> {
+    // Handle whiteout conversion
+    let name = path.file_name().unwrap();
+    let parent = path.parent().unwrap();
+    let name = name.to_str().unwrap();
+    if name == ".wh..wh..opq" {
+        xattr::set(parent, "trusted.overlay.opaque", b"y")?;
+        return Ok(false)
+    }
+
+    if name.starts_with(".wh.") {
+        let original_name = name.strip_prefix(".wh.").unwrap();
+        let original_path = parent.join(original_name);
+        let path = CString::new(format!(
+            "{}/{}",
+            destination.display(),
```

Comment:
In rust edition >=2021 (guest components/image meets this) due to https://doc.rust-lang.org/std/fmt/trait.Display.html
```rust
format!(
    "{}/{}",
    destination.display(),
    original_path.display(),
}
```
can be shortten to
```rust
format!("{destination}/{original_path}")
```

Reply:
nice!

Reply:
hmm... I ended up reverting this as I was getting...
```
error[E0277]: `Path` doesn't implement `std::fmt::Display`
   --> image-rs/src/unpack.rs:106:42
    |
106 |         let path = CString::new(format!("{destination}/{original_path}"))?;
    |                                          ^^^^^^^^^^^^^ `Path` cannot be formatted with the default formatter; call `.display()` on it
    |
    = help: the trait `std::fmt::Display` is not implemented for `Path`, which is required by `&Path: std::fmt::Display`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: call `.display()` or `.to_string_lossy()` to safely print paths, as they may contain non-Unicode data
    = note: this error originates in the macro `$crate::__export::format_args` which comes from the expansion of the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `PathBuf` doesn't implement `std::fmt::Display`
   --> image-rs/src/unpack.rs:106:56
    |
106 |         let path = CString::new(format!("{destination}/{original_path}"))?;
    |                                                        ^^^^^^^^^^^^^^^ `PathBuf` cannot be formatted with the default formatter; call `.display()` on it
    |
    = help: the trait `std::fmt::Display` is not implemented for `PathBuf`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: call `.display()` or `.to_string_lossy()` to safely print paths, as they may contain non-Unicode data
    = note: this error originates in the macro `$crate::__export::format_args` which comes from the expansion of the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)
```